woocommerce-openpix.php

```diff
@@ -266,3 +278,44 @@ public function load_plugin_assets()
         wp_enqueue_style('openpix_frontend_css');
     }
 }
+
+function checkCompabilityCheckoutBlock()
+{
+    $pluginId = wc_get_container()
+        ->get(\Automattic\WooCommerce\Utilities\PluginUtil::class)
+        ->get_wp_plugin_id(__FILE__);
+
+    return is_plugin_active($pluginId) &&
+        class_exists('Automattic\WooCommerce\Blocks\Package') &&
+        file_exists(
+            filename: __DIR__ .
+                '/assets/' .
+                WC_OpenPix_Pix_Block::PIX_BLOCK_SCRIPT_FILENAME
+        );
+}
+
+function userNoticeIncompatibilityWithCheckoutBlock()
+{
+    if (checkCompabilityCheckoutBlock()) {
+        return false;
+    }
+
+    //     echo <<<HTML
+    //     <div clas    s="woocommerce-info">
+    //         Atenção: O plugin "woocommerce-openpix" ainda não oferece suporte ao Checkout Blocks. Então estamos usando o modelo de checkout classico.
+    //     </div>
+    // HTML;
+}
+
+function adminNoticeIncompatibilityWithBlock()
+{
+    if (checkCompabilityCheckoutBlock()) {
+        return false;
+    }
+
+    echo <<<HTML
+    <div class="notice notice-warning">
+        <p><strong>woocommerce-openpix</strong> ainda não é compatível com o Checkout de Blocos do WooCommerce. Para evitar problemas, utilize o Checkout Classico.</p>
```

Comment:
is this correct ?

Reply:
Yes, it is similar to a string, heredoc can allow indentation, line wrapping and variable interpolation

This code has the same effect as this one
```php
echo '<div class="notice notice-warning"><p><strong>woocommerce-openpix</strong> ainda não é compatível com o Checkout de Blocos do WooCommerce. Para evitar problemas, utilize o Checkout Classico.</p></div>';
```

Ref:
https://www.php.net/manual/pt_BR/language.types.string.php#language.types.string.syntax.heredoc